File: services/data/indexes.py
import os
from dotenv import load_dotenv
from pymongo import MongoClient

load_dotenv()

MONGO_URI = os.getenv("MONGO_URI", "mongodb://localhost:27017/traffic_system")

mongo = MongoClient(MONGO_URI)
db = mongo["traffic_system"]

try:
    db["regions"].create_index("regionId", unique=True)
    db["region_graph"].create_index("regionId", unique=True)
    db["users"].create_index("userId", unique=True)
    db["vehicles"].create_index("vehicleId", unique=True)
    db["bookings"].create_index("bookingId", unique=True)
    db["bookings"].create_index("requestId", unique=True)

    print("✅ Indexes created successfully")
except Exception as e:
    print("❌ Failed to create indexes")
    print(e)

File: scripts/indexes.js
require("dotenv").config();
const { MongoClient } = require("mongodb");

async function createIndexes() {
  const client = new MongoClient(process.env.MONGO_URI);

  try {
    await client.connect();
    const db = client.db("traffic_system");

    await db.collection("users").createIndex({ userId: 1 }, { unique: true });
    await db.collection("vehicles").createIndex({ vehicleId: 1 }, { unique: true });
    await db.collection("bookings").createIndex({ bookingId: 1 }, { unique: true });
    await db.collection("bookings").createIndex({ requestId: 1 }, { unique: true });
    await db.collection("bookings").createIndex({ vehicleId: 1 });
    await db.collection("regions").createIndex({ regionId: 1 }, { unique: true });

    console.log("✅ Indexes created successfully");
  } catch (err) {
    console.error("❌ Failed to create indexes");
    console.error(err);
  } finally {
    await client.close();
  }
}

createIndexes();